src/rgw/rgw_keystone.cc

```diff
@@ -136,6 +136,35 @@ std::string CephCtxConfig::get_admin_password() const noexcept  {
   return empty;
 }
 
+#define SMALL_TIME_SLOP 300
+#define BIG_TIME_SLOP 86400
+int Service::validate_admin_token(TokenEnvelope& t)
+{
+  const uint64_t now = ceph_clock_now().sec();
+  int ret = ~0xffff;
+  if (t.token.id.empty()) {
+    ret |= 1;
```

Comment:
it looks like this bit twiddling stuff is just for debug logging, but dumping the `flags` alone won't help a user diagnose these errors. can we move the logging into this function, so we can print a useful error message for each case?